File: src/main/java/com/cloudforce/service/PriceBookService.java
package com.cloudforce.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.cloudforce.domain.PriceBook;
import com.cloudforce.dto.PriceBookDTO;
import com.cloudforce.dto.PriceBookSearchDTO;
import com.cloudforce.dto.PriceBookPageDTO;
import com.cloudforce.dto.PriceBookConvertCriteriaDTO;
import com.cloudforce.service.GenericService;
import com.cloudforce.dto.common.RequestDTO;
import com.cloudforce.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PriceBookService extends GenericService<PriceBook, Integer> {

	List<PriceBook> findAll();

	ResultDTO addPriceBook(PriceBookDTO priceBookDTO, RequestDTO requestDTO);

	ResultDTO updatePriceBook(PriceBookDTO priceBookDTO, RequestDTO requestDTO);

    Page<PriceBook> getAllPriceBooks(Pageable pageable);

    Page<PriceBook> getAllPriceBooks(Specification<PriceBook> spec, Pageable pageable);

	ResponseEntity<PriceBookPageDTO> getPriceBooks(PriceBookSearchDTO priceBookSearchDTO);
	
	List<PriceBookDTO> convertPriceBooksToPriceBookDTOs(List<PriceBook> priceBooks, PriceBookConvertCriteriaDTO convertCriteria);

	PriceBookDTO getPriceBookDTOById(Integer priceBookId);


}

File: cloudforce_ui/src/services/PriceBookService.js
import http from "../http-common"; 

class PriceBookService {
  getAllPriceBooks(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/priceBook/priceBooks`, searchDTO);
  }
 

  get(priceBookId) {
    return this.getRequest(`/priceBook/${priceBookId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/priceBook?field=${matchData}`, null);
  }

  addPriceBook(data) {
    return http.post("/priceBook/addPriceBook", data);
  }

  update(data) {
  	return http.post("/priceBook/updatePriceBook", data);
  }
  
  uploadImage(data,priceBookId) {
  	return http.postForm("/priceBook/uploadImage/"+priceBookId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PriceBookService();
